### ISSUE TITLE
TypeError bug started v0.2.30

### ISSUE THREAD

**Issue by Josh-XT** [abetlen/llama-cpp-python#1107]
I am getting an error since updating to v0.2.30, it goes away when I revert to v0.2.29.

Error below:

```
 File "C:\Users\Josh\Repos\AGiXT\Local-LLM\local_llm\__init__.py", line 368, in chat
    data = self.generate(prompt=prompt)
  File "C:\Users\Josh\Repos\AGiXT\Local-LLM\local_llm\__init__.py", line 326, in generate
    data = Llama(**self.params).create_completion(
  File "C:\Users\Josh\AppData\Local\Programs\Python\Python310\lib\site-packages\llama_cpp\llama.py", line 1305, in create_completion
    completion: Completion = next(completion_or_chunks)  # type: ignore
  File "C:\Users\Josh\AppData\Local\Programs\Python\Python310\lib\site-packages\llama_cpp\llama.py", line 831, in _create_completion
    for token in self.generate(
  File "C:\Users\Josh\AppData\Local\Programs\Python\Python310\lib\site-packages\llama_cpp\llama.py", line 606, in generate
    token = self.sample(
  File "C:\Users\Josh\AppData\Local\Programs\Python\Python310\lib\site-packages\llama_cpp\llama.py", line 524, in sample
    id = self._ctx.sample_token_mirostat_v2(
  File "C:\Users\Josh\AppData\Local\Programs\Python\Python310\lib\site-packages\llama_cpp\_internals.py", line 426, in sample_token_mirostat_v2
    return llama_cpp.llama_sample_token_mirostat_v2(
  File "C:\Users\Josh\AppData\Local\Programs\Python\Python310\lib\site-packages\llama_cpp\llama_cpp.py", line 2320, in llama_sample_token_mirostat_v2
    return _lib.llama_sample_token_mirostat_v2(ctx, candidates, tau, eta, mu)
ctypes.ArgumentError: argument 5: TypeError: expected LP_c_float instance instead of float
```

**Comment by abetlen** [comment]
@Josh-XT sorry about that, should be fixed now in v0.2.31, additionally mirostat sampling has been kinda broken #914 but that should be fixed now too.

**Comment by Josh-XT** [comment]
> @Josh-XT sorry about that, should be fixed now in v0.2.31, additionally mirostat sampling has been kinda broken #914 but that should be fixed now too.

No problem, you're awesome! Thank you!
